### ISSUE TITLE
Pretty print type signatures

### ISSUE THREAD

**Issue by coot** [haskell/haddock#1405]
It would be nice to have a pretty printer for type signatures in haddock.  Complex types are not very readable, and often don't even fit into a single screen line.

**Comment by sjakobi** [comment]
Duplicate of https://github.com/haskell/haddock/issues/472?

**Comment by coot** [comment]
I don't think so. #472 only applies to type signatures which have haddoc annotations, which are pretty printed (at least split into separate lines), an [example](https://hackage.haskell.org/package/free-algebras-0.1.0.1/docs/Control-Algebra-Free2.html#v:assocFree2) of type signature that is not pretty printed.
